plugins/more_selectors/more_selectors.py

```diff
@@ -0,0 +1,201 @@
+        if isinstance(axis_value, str):
+            try:
+                return named_vectors[axis_value]
+            except KeyError:
+                raise ValueError("Supported names are {}".format(list(named_vectors.keys())))    
+        else:
+            if cq.Vector(axis_value).Length != 0:
+                return cq.Vector(axis_value)
+            else:
+                raise ValueError("Axis of the cylinder must be non null")
+
+    def get_ortho_vector(self, vector):
+        #returns a vector orthogonal to the provided vector
+        if 0 not in vector.toTuple():
+            return (1,1, (-vector.x - vector.y)/vector.z)    
+        else:
+            if vector.x != 0:
+                return ((-vector.y - vector.z)/vector.x, 1, 1)
+            elif vector.y != 0:
+                return (1, (-vector.x - vector.z)/vector.y,1)
+            else:
+                return (1,1, (-vector.x - vector.y)/vector.z)
+
+    def filter(self, objectList):
+        result =[]
+        for o in objectList:            
+            p = o.Center()
+            projected_p = self.base.toLocalCoords(p)
+            p_radius = sqrt(projected_p.x**2 + projected_p.y**2)
+
+            if p_radius < self.outer_radius:   
+                result.append(o)
+                    
+
+        return result
+
+class InfHollowCylinderSelector(InfiniteCylinderSelector):
+    """
+    Selects any shape present in the defined infinite hollow  
+    cylinder based on the shape center of mass point.   
+    """
+    def __init__(self, origin, along_axis, outer_radius, inner_radius, debug=False):
+        super().__init__(origin, along_axis, outer_radius, debug=False)
+        self.inner_radius = inner_radius
+        if debug:
+            make_debug_cylinder(self.base, self.outer_radius, inner_radius = self.inner_radius)
+    def filter(self, objectList):
+        result =[]
+        for o in objectList:            
+            p = o.Center()
+            projected_p = self.base.toLocalCoords(p)
+            p_radius = sqrt(projected_p.x**2 + projected_p.y**2)
+
+            if p_radius < self.outer_radius and p_radius > self.inner_radius:   
+                result.append(o)   
+        return result
+
+class CylinderSelector(InfiniteCylinderSelector):
+    """
+    Selects any shape present in the defined cylinder 
+    based on the shape center of mass point.   
+    """
+    def __init__(self, origin, along_axis, height, radius, debug=False):
+        super().__init__(origin, along_axis, radius)
+        self.height = height
+        if debug:
+            make_debug_cylinder(self.base, radius, height=height)
+
+    def filter(self, objectList):
+        result =[]
+        for o in objectList:            
+            p = o.Center()
+            projected_p = self.base.toLocalCoords(p)
+            p_radius = sqrt(projected_p.x**2 + projected_p.y**2)
+
+            if p_radius < self.outer_radius and (projected_p.z < self.height and projected_p.z > 0):   
```

Comment:
Do you mean something like this : 
`test = cq.Workplane("XY").box(5,5,5)
sel = test.vertices(CylinderSelector((0,0,-2), "Z",5,10))`

If yes then I'm not sure why you think it wouldn't work as intended. This example works. Since the tested point are projected into the base plan local coordinated system it won't care if it has negative global Z coord. Or maybe I'm missing something?

Reply:
Yeah, that good. I just wanted to make sure.